File: app/src/main/java/co/early/password123/feature/networkmonitor/ActivityLifecycleNetworkCheck.java
package co.early.password123.feature.networkmonitor;


import android.app.Activity;
import android.app.Application.ActivityLifecycleCallbacks;
import android.os.Bundle;

import co.early.asaf.core.Affirm;


public class ActivityLifecycleNetworkCheck implements ActivityLifecycleCallbacks {

    private final NetworkState networkState;

    public ActivityLifecycleNetworkCheck(NetworkState networkState) {
        this.networkState = Affirm.notNull(networkState);
    }

    @Override
    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {

    }

    @Override
    public void onActivityStarted(Activity activity) {

    }

    @Override
    public void onActivityResumed(Activity activity) {
        networkState.checkIfDown();
    }

    @Override
    public void onActivityPaused(Activity activity) {

    }

    @Override
    public void onActivityStopped(Activity activity) {

    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {

    }

    @Override
    public void onActivityDestroyed(Activity activity) {

    }
}

File: app/src/main/java/co/early/password123/feature/networkmonitor/ActivityLifecycleNetworkCheck.kt
package co.early.password123.feature.networkmonitor


import android.app.Activity
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle

import co.early.asaf.core.Affirm


class ActivityLifecycleNetworkCheck(networkState: NetworkState) : ActivityLifecycleCallbacks {

    private val networkState: NetworkState

    init {
        this.networkState = Affirm.notNull(networkState)
    }

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {

    }

    override fun onActivityStarted(activity: Activity) {

    }

    override fun onActivityResumed(activity: Activity) {
        networkState.checkIfDown()
    }

    override fun onActivityPaused(activity: Activity) {

    }

    override fun onActivityStopped(activity: Activity) {

    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle?) {

    }

    override fun onActivityDestroyed(activity: Activity) {

    }
}
